lib/src/ImmutableSet.ts

```diff
@@ -0,0 +1,51 @@
+import { Comparable, comparableEquals } from './Comparable';
+import { AbstractSet } from './AbstractSet';
+import { ArrayUtils } from './ArrayUtils';
+
+export class ImmutableSet<T extends Comparable> extends AbstractSet<T> {
```

Comment:
rename ImmutableSet to Set, merge AbstractSet et Set ensemble ?

Reply:
AbstractSet n'a d'abstrait que son nom, c'est un set très concret :p Puis vu qu'on a plus de MutableSet, autant tout mettre au même endroit !